R/neatsort.R

```diff
@@ -0,0 +1,186 @@
+    }
+    
+    # Check for unique column names
+    if (any(duplicated(colnames(x)))) {
+        stop("Column names of the matrix must be unique.")
+    }
+}
+
+
+#' @description Sorts a matrix by radial theta angle.
+#' @param x A matrix containing the ordinated data to be sorted.
+#' @param subset A vector specifying a subset of rows to be used and retained.
+#' @param dimensions A vector of length 2 specifying the columns of the matrix to use for the X and Y coordinates.
+#' @param centering_method A character string specifying the method to center the data.
+#' @param sorting_order A character string specifying the order of sorting.
+#' @return A vector of row names in the sorted order.
+#' @rdname neatsort
+#' @export
+setMethod("neatsort", signature = c("matrix"),
+          function(x,
+                   subset = NULL,
+                   dimensions = c(1, 2),
+                   centering_method = c("mean", "median", "none"),
+                   sorting_order = c("ascending", "descending"),
+                   ...){
+              
+                # Check args
+                .check_neatsort_args(x, subset, dimensions, centering_method, sorting_order)
+              
+                # Create subset if required
+                if( !is.null(subset) ){
+                    x <- .take_subset(x, subset)
+                }
+
+                # Take the correct dimensions
+                x <- .take_dimensions(x, dimensions)
+
+                # Get the theta values and order them
+                theta_values <- .radial_theta(x, centering_method)
+                ordering <- .get_sorted_rownames(theta_values, sorting_order)
+
+                return(ordering)
+          }
+    )
+
+
+# Takes a subset of rows from the data matrix.
+.take_subset <- function(data, subset) {
+    data <- data[subset, , drop = FALSE]
+    return(data)
+}
+
+
+# Takes the specified columns (dimensions) from the data matrix.
+.take_dimensions <- function(data, dimensions) {
+    data <- data[, dimensions, drop = FALSE]
+    return(data)
+}
+
+
+# Computes the radial theta values for each row in the data matrix.
+.radial_theta <- function(data, centering_method) {
+    if (centering_method == "mean") {
+        centered_data <- scale(data, center = TRUE, scale = FALSE)
+    } else if (centering_method == "median") {
+        centered_data <- scale(data, center = apply(data, 2, median), scale = FALSE)
+    } else if (centering_method == "none") {
+        centered_data <- data
+    } else {
+        stop("Unsupported centering method. Choose either 'mean', 'median', 'mode', or 'none'.")
+    }
+    
+    theta <- atan2(centered_data[, 2], centered_data[, 1])
+    names(theta) <- rownames(centered_data)
+    
+    return(theta)
+}
+
+# Sorts the theta values and returns the ordered row names.
+.get_sorted_rownames <- function(theta_values, sorting_order) {
+    sorted_indices <- order(theta_values, decreasing = (sorting_order == "descending"))
```

Comment:
You could rename the parameter to decreasing which get values TRUE or FALSE. That would make the code simpler. Also as the same parameter is used in base R, it would make this more intuitive